Jacob distancebetween

cuthbert1221 opened the pull request:
Adding the CSV Writer Module
This branch still has the unmerged changes from adding the tests for the geodist.py 
I can move these out if required.

cuthbert1221 committed:
Adding Distance-between Module

moved geodist.py into distance-between module.
Added tests to geodist.py
Edited geodist.py to retuen with -1 if no co-ords are suplied.

Commit:
Adding CSV File Writer

Added the CSV writer method, plus 5 tests.

ashketchupppp reviewed (changes requested):
What you have here is exactly what I was hoping for, well done.

Perhaps we could merge this module with the csvreader module to make one "fileutil" module, seems a better idea than splitting them up. Not sure why we split them up in the first place.

Like I said in the slack message, the csvwriter test is testing two things: correct file reading and correct conversion of csv formatted strings to a list of dictionaries. Implicitly testing two things in one test is bad practice because if the test fails then you don't know which thing caused the test failure. Could you write a file writer function (make a new file for it and put it in the new fileutil module) and then use that to write your file? That way you can test that part and you are 100% sure it works when you use it in csvwriter. 

Approving with the requested changes, good work

cuthbert1221 commented:
Yes, I'll add the file writer function over the weekend.

One question, when you say "Approving with the requested changes"
Does this mean merging the current request and then add changes in the new one? Or wait until amendments then merge all together,

Commit:
File Module

FileUtil + tests
Writer + tests

Commit:
CSV Module

Added CSV:
Two Main fucntions:
1. csv_to_dict   requires filepath, dict of content

2. dict_to_csv   requires filepath
